### PR TITLE
[Nuclio] Delete remote function config map

### PR DESCRIPTION
**Pull request by rokatyy**
Jira - https://iguazio.atlassian.net/browse/ML-6394

### CODE REVIEW

**Review by TomerShor — COMMENTED** [review]
Looks good!
Add it to a unit test and ensure was the k8s helper's methods were called

**Review by TomerShor — APPROVED** [review]
LGTM 😸

File: server/api/utils/singletons/k8s.py [hunk 1]
```diff
@@ -552,28 +552,22 @@ def ensure_configmap(
...
-        name = (
-            full_name
-            if len(full_name) <= 63
-            else full_name[:59] + self._generate_rand_string(4)
-        )
 
-        have_confmap = False
-        configmaps_with_label = self.v1api.list_namespaced_config_map(
-            namespace=namespace, label_selector=f"{label_name}={full_name}"
+        configmap_with_label = self.get_configmap_for_function(
```
Comment (yanburman) [inline server/api/utils/singletons/k8s.py]: this function is generic right now and not specifically bound to functions. This code changes those semantics

Comment (rokatyy) [reply server/api/utils/singletons/k8s.py]: @yanburman agreed! Changed naming

**Review by liranbg — APPROVED** [review]

**Pull request merged**